hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerResponseFieldsTest.kt

```diff
@@ -247,4 +247,200 @@ class PrisonerResponseFieldsTest : AbstractSearchIntegrationTest() {
         .returnResult().responseBody!!
     }
   }
+
+  @DisplayName("Find by prison search")
+  @Nested
+  inner class FindByPrisonSearch {
+    @Test
+    fun `should return bad request for unknown fields`() {
+      webTestClient.searchError(
+        prisonId = "MDI",
+        responseFields = listOf("prisonerNumber", "doesNotExist", "nested.alsoDoesNotExist"),
+      ).also {
+        assertThat(it.userMessage).doesNotContain("prisonerNumber")
+        assertThat(it.userMessage).contains("doesNotExist")
+        assertThat(it.userMessage).contains("nested.alsoDoesNotExist")
+      }
+    }
+
+    @Test
+    fun `should return fields of different types`() {
+      val response = webTestClient.search(
+        prisonId = "MDI",
+        responseFields = listOf(
+          "prisonerNumber",
+          "dateOfBirth",
+          "recall",
+          "heightCentimetres",
+          "currentIncentive.level.description",
+        ),
+      )
+
+      with(response.content.first()!!) {
+        assertThat(prisonerNumber).isEqualTo("A1234AA")
+        assertThat(dateOfBirth).isEqualTo("1975-07-20")
+        assertThat(recall).isEqualTo(true)
+        assertThat(heightCentimetres).isEqualTo(180)
+        assertThat(currentIncentive?.level?.description).isEqualTo("Standard")
+      }
+    }
```

Comment:
I've just copied all the same tests from the other endpoint I applied this change to. I don't think every endpoint needs all of these tests because they're essentially testing the same thing - that we can request objects, nested fields etc.

However I'm implementing all tests here just for expediency, will probably remove duplicate when applying to further endpoints.